src/TabBar.tsx

```diff
@@ -59,6 +60,13 @@ export type Props<T extends Route> = SceneRendererProps & {
 
 type State = {
   layout: Layout;
+  tabWidths: number[];
+};
+
+const allDone = (arr: number[], total: number) => {
+  return Array.from({ length: total }).reduce<boolean>((result, _, i) => {
```

Comment:
yes!

Reply:
I am afraid, this will not be correct. Because let's say 2nd tab's onLayout fires first, then length will be 2 but the array has only one element, this will give us false positives. I've changed this to a regular for-loop to make it more readable.

Reply:
that's what `widths.every(Boolean)` is for

Reply:
```js
const widths = [];
// assuming third tab's onLayout fires first
widths[3] = 24
// now widths is an array but has only one item present at index=3
// but JS  => widths.length = 4 and it will satisfy widths.length === length
// and it'll satisfy widths.every(Boolean) as every will run only for one item present at index=3 and not for index=0,1,2


```
widths.every(Boolean)
true

So we can't rely on this as it'll give false positive everytime.

Reply:
Hmm, I think we just need an `Array.from` to convert the holes to proper array:

```js
widths.length === total && Array.from(widths).every(Boolean)
```